File: Codechef/remone.py
#include <bits/stdc++.h>
#define IOS ios_base::sync_with_stdio(false); cin.tie(NULL);
#define loop(i,n) for(int i=0;i<n;i++)
#define for1(i,n) for(int i=1;i<=n;i++)
typedef long long ll;
using namespace std;
void solve(){ 
    ll n;
    cin>>n;
    vector<int>arr(n),brr(n-1);
    loop(i,n)cin>>arr[i];
    loop(i,n-1)cin>>brr[i];
    vector<ll> freq(10000,0);
    unordered_map<ll,ll> map;
    for(int i=0;i<n;i++){
        for (int j = 0; j <n-1; j++)
        {
             map[brr[i]-arr[j]]++;
        } 
    }
    ll ans=INT_MIN;
    for(auto x : map){
        if(x.second>ans){
            ans=x.first;
        }
    }
     for (auto x : map)
      cout << x.first << " " << x.second << endl;
    
    cout<<ans<<endl;
}
int main() {
int t=1;
cin>>t;
while(t--){
 solve(); 
}
     return 0;
}

File: Codechef/remone.cpp
#include <bits/stdc++.h>
#define IOS ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
#define loop(i,a,n) for(int i=a;i<n;i++)
#define all(v) (v).begin(),(v).end()
#define all1(v) (v).begin()+1,(v).end()
#define allr(v) (v).rbegin(),(v).rend()
#define allr1(v) (v).rbegin()+1,(v).rend()
typedef long long ll;
const ll INF = 1e9+7;
const ll MOD = 998244353;
using namespace std;
void solve(){ 
      ll n;
      cin>>n;
      vector<ll>a(n),b(n-1);
      unordered_set<ll>s;
      loop(i,0,n)cin>>a[i],s.insert(a[i]);
      loop(i,0,n-1)cin>>b[i];
      sort(all(a));
      sort(all(b));
      ll ans = b[0]-a[1];
      for(auto i : b){
           if(s.count(i-ans)==0){
                ans = b[0]-a[0];
                break;
           }
      }
      if (ans<=0) cout<<b[0]-a[0]<<endl;
      else cout<<ans<<endl;
}
int main() {
int t=1;
cin>>t;
while(t--){
 solve(); 
}
      return 0;
}